tsk/img/img_writer.cpp

```diff
@@ -801,9 +830,23 @@ TSK_RETVAL_ENUM tsk_img_writer_create(TSK_IMG_INFO *img_info, const TSK_TCHAR *o
     writer->nextDataOffset = 0x600 + batLengthOnDisk;
 
     /* Initialize all the bookkeeping arrays */
-    writer->blockStatus = (IMG_WRITER_BLOCK_STATUS_ENUM*)tsk_malloc(writer->totalBlocks * sizeof(IMG_WRITER_BLOCK_STATUS_ENUM));
-    writer->blockToSectorNumber = (uint32_t*)tsk_malloc(writer->totalBlocks * sizeof(uint32_t));
-    writer->blockToSectorBitmap = (unsigned char **)tsk_malloc(writer->totalBlocks * sizeof(unsigned char *));
+    if ((writer->blockStatus = (IMG_WRITER_BLOCK_STATUS_ENUM*)tsk_malloc(writer->totalBlocks * sizeof(IMG_WRITER_BLOCK_STATUS_ENUM))) == NULL) {
+        free(writer->fileName);
+        free(raw_info->img_writer);
+        return TSK_ERR;
+    }
+    if ((writer->blockToSectorNumber = (uint32_t*)tsk_malloc(writer->totalBlocks * sizeof(uint32_t))) == NULL) {
+        free(writer->blockStatus);
+        free(writer->fileName);
+        free(raw_info->img_writer);
+        return TSK_ERR;
+    }
+    if ((writer->blockToSectorBitmap = (unsigned char **)tsk_malloc(writer->totalBlocks * sizeof(unsigned char *))) == NULL) {
+        free(writer->blockToSectorNumber);
```

Comment:
Should you also free blockStatus?

Reply:
yes.
I should not free blockToSectorNumber, it is already NULL.